server/src/models/user.model.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  AllowNull,
+  Column,
+  DataType,
+  Default,
+  HasMany,
+  IsUUID,
+  Model,
+  PrimaryKey,
+  Table,
+  Unique,
+} from 'sequelize-typescript';
+import UserAddress from './userAddress.model';
+
+@Table({
+  tableName: 'user',
+  freezeTableName: true,
+  timestamps: true,
+  createdAt: 'created_at',
+  updatedAt: 'updated_at',
+})
+export default class User extends Model {
+  @IsUUID(4)
+  @PrimaryKey
+  @Default(DataType.UUIDV4)
+  @Column
+  id!: string;
+
+  @Unique
+  @AllowNull(false)
+  @Column(DataType.STRING(30))
+  login_id!: string;
+
+  @Unique
+  @AllowNull(false)
+  @Column(DataType.STRING(20))
+  email!: string;
+
+  @AllowNull(false)
+  @Column(DataType.STRING(20))
+  name!: string;
+
+  @AllowNull(true)
+  @Column(DataType.STRING(100))
+  password: string | undefined; // oauth시 password 없음
+
+  @AllowNull(false)
+  @Column(DataType.STRING(12))
+  phone!: string;
+
+  @AllowNull(true)
+  @Column(DataType.STRING(10))
+  provider: string | undefined;
+
+  @HasMany(() => UserAddress)
+  user_address: UserAddress[] | undefined;
+
+  @AllowNull(true)
+  @Column(DataType.INTEGER)
+  default_address: number | undefined;
```

Comment:
```suggestion
  provider?: string;

  @HasMany(() => UserAddress)
  user_address?: UserAddress[];

  @AllowNull(true)
  @Column(DataType.INTEGER)
  default_address?: number;
```

이렇게 표현할 수도 있을 거 같아요 😀